src/lib/rosetta_coding/coding.ml

```diff
@@ -104,183 +111,52 @@ end
 let pack (type t) (module M : Unpacked with type t = t) raw =
   (* 256 bits = 64 hex chars *)
   assert (Int.equal (String.length raw) 64) ;
-  let bits0 =
+  let bits =
     String.to_list raw |> List.map ~f:hex_char_to_bits4 |> List.concat
   in
-  (* remove padding bit *)
-  let bits = List.tl_exn bits0 |> List.rev in
-  M.project bits
+  (* In our encoding, we have highest bytes at the end and lowest at the
+     beginning. *)
+  let bytes = bits_by_8s bits in
+  let bytes' = List.rev bytes in
+  let bits' = List.concat bytes' in
 
-let to_field = pack (module Field)
+  let padding_bit = List.hd_exn bits' in
+  (* remove padding bit *)
+  let bits'' = List.tl_exn bits' |> List.rev in
+  (padding_bit, M.project bits'')
```

Comment:
do we use the returned padding_bit?

is it worth asserting that the padding bit is `false`?

Reply:
sometimes it's not false -- the "compressed pk encoding" uses that top bit for is_odd

```
let to_public_key_compressed raw =
  let is_odd, x = pack (module Field) raw in
  { Public_key.Compressed.Poly.x; is_odd }
```